crates/router/src/core/payment_methods/cards.rs

```diff
@@ -261,6 +262,310 @@ pub async fn get_or_insert_payment_method(
+                            expiry_year: Some(card_details.card_exp_year.clone()),
+                            card_token: None,
+                            card_fingerprint: None,
+                            card_holder_name: card_details.card_holder_name.clone(),
+                            nick_name: card_details.nick_name.clone(),
+                            card_isin: Some(card_isin.clone()),
+                            card_issuer: card_details.card_issuer.clone().or(card_info.card_issuer),
+                            card_network: card_details
+                                .card_network
+                                .clone()
+                                .or(card_info.card_network),
+                            card_type: card_details.card_type.clone().or(card_info.card_type),
+                            saved_to_locker: false,
+                        }
+                    })
+                    .unwrap_or_else(|| {
+                        logger::debug!("Failed to fetch bin details");
+                        api::CardDetailFromLocker {
+                            scheme: card_details
+                                .card_network
+                                .clone()
+                                .map(|card_network| card_network.to_string()),
+                            last4_digits: Some(last4_digits.clone()),
+                            issuer_country: card_details.card_issuing_country.clone(),
+                            card_number: None,
+                            expiry_month: Some(card_details.card_exp_month.clone()),
+                            expiry_year: Some(card_details.card_exp_year.clone()),
+                            card_token: None,
+                            card_fingerprint: None,
+                            card_holder_name: card_details.card_holder_name.clone(),
+                            nick_name: card_details.nick_name.clone(),
+                            card_isin: Some(card_isin.clone()),
+                            card_issuer: card_details.card_issuer.clone(),
+                            card_network: card_details.card_network.clone(),
+                            card_type: card_details.card_type.clone(),
+                            saved_to_locker: false,
+                        }
+                    }),
+            )
+        }
+    } else {
+        None
+    };
+
+    let payment_method_card_details = card
+        .as_ref()
+        .map(|card| PaymentMethodsData::Card(CardDetailsPaymentMethod::from(card.clone())));
+
+    let payment_method_data_encrypted =
+        create_encrypted_data(key_store, payment_method_card_details)
+            .await
+            .map(|details| details.into());
+
+    let payment_method_metadata: Option<serde_json::Value> =
+        req.metadata.as_ref().map(|data| data.peek()).cloned();
+
+    let payment_method_id = generate_id(consts::ID_LENGTH, "pm");
+
+    let client_secret = generate_id(
```

Comment:
why we are creating client_secret here?

Reply:
This is change is form pr (https://github.com/juspay/hyperswitch/pull/4134). Where in which the generated `client_secret` will be stored in db